File: Environment-Interactive Agent Pattern/langgraph_python_example/environment_interactive_agent.py
import os
import requests

def ask_mistral(messages):
    api_key = os.getenv('MISTRAL_API_KEY')
    if not api_key:
        raise ValueError('Please set MISTRAL_API_KEY in your .env file')
    url = 'https://api.mistral.ai/v1/chat/completions'
    headers = {
        'Authorization': f'Bearer {api_key}',
        'Content-Type': 'application/json',
    }
    data = {
        'model': 'mistral-tiny',
        'messages': messages,
    }
    response = requests.post(url, headers=headers, json=data)
    response.raise_for_status()
    return response.json()['choices'][0]['message']['content'].strip()

def main():
    env_state = input('Describe the current environment state: ')
    messages = [
        {'role': 'system', 'content': 'You are an environment-interactive agent. Perceive the environment, choose an action, and adapt based on feedback.'},
        {'role': 'user', 'content': f'Environment state: {env_state}'},
    ]
    action = ask_mistral(messages)
    print('\nAgent Action:\n', action)

    feedback = input('Describe the environment feedback after the action: ')
    feedback_messages = [
        {'role': 'system', 'content': 'You are an environment-interactive agent. Given the feedback, adapt your strategy or next action.'},
        {'role': 'user', 'content': f'Feedback: {feedback}'},
    ]
    adaptation = ask_mistral(feedback_messages)
    print('\nAgent Adaptation/Next Action:\n', adaptation)

if __name__ == '__main__':
    main()

File: Environment-Interactive Agent Pattern/autogen_typescript_example/environment_interactive_agent.ts
import dotenv from 'dotenv';
dotenv.config();
import axios from 'axios';
import readline from 'readline';

const MISTRAL_API_KEY = process.env.MISTRAL_API_KEY;
const MISTRAL_API_URL = 'https://api.mistral.ai/v1/chat/completions';

if (!MISTRAL_API_KEY) {
  console.error('Please set MISTRAL_API_KEY in your .env file');
  process.exit(1);
}

async function askMistral(messages: any[]) {
  const response = await axios.post(
    MISTRAL_API_URL,
    {
      model: 'mistral-tiny',
      messages,
    },
    {
      headers: {
        'Authorization': `Bearer ${MISTRAL_API_KEY}`,
        'Content-Type': 'application/json',
      },
    }
  );
  return response.data.choices[0].message.content.trim();
}

async function main() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  rl.question('Describe the current environment state: ', async (envState) => {
    let messages = [
      { role: 'system', content: 'You are an environment-interactive agent. Perceive the environment, choose an action, and adapt based on feedback.' },
      { role: 'user', content: `Environment state: ${envState}` },
    ];

    let action = await askMistral(messages);
    console.log('\nAgent Action:\n', action);

    rl.question('Describe the environment feedback after the action: ', async (feedback) => {
      let feedbackMessages = [
        { role: 'system', content: 'You are an environment-interactive agent. Given the feedback, adapt your strategy or next action.' },
        { role: 'user', content: `Feedback: ${feedback}` },
      ];
      let adaptation = await askMistral(feedbackMessages);
      console.log('\nAgent Adaptation/Next Action:\n', adaptation);
      rl.close();
    });
  });
}

main();
